src/TerraformTaintPlugin.groovy

```diff
@@ -0,0 +1,121 @@
+import static TerraformEnvironmentStage.PLAN_COMMAND
+
+class TerraformTaintPlugin implements TerraformEnvironmentStagePlugin, TerraformTaintCommandPlugin, TerraformUntaintCommandPlugin, Resettable {
+    public static String origin_repo = ''
+    private static DEFAULT_BRANCHES = ['master']
+    private static branches = DEFAULT_BRANCHES
+
+    public static void init() {
+        TerraformTaintPlugin plugin = new TerraformTaintPlugin()
+
+        BuildWithParametersPlugin.withStringParameter([
+            name: "TAINT_RESOURCE",
+            description: 'Run `terraform taint` on the resource specified prior to planning and applying.'
+        ])
+
+        BuildWithParametersPlugin.withStringParameter([
+            name: "UNTAINT_RESOURCE",
+            description: 'Run `terraform untaint` on the resource specified prior to planning and applying.'
+        ])
+
+        TerraformEnvironmentStage.addPlugin(plugin)
+        TerraformTaintCommand.addPlugin(plugin)
+        TerraformUntaintCommand.addPlugin(plugin)
+    }
+
+    public static onlyOnOriginRepo(String origin_repo) {
+        this.origin_repo = origin_repo
+        return this
+    }
+
+    public static onMasterOnly() {
+        this.branches = ['master']
+        return this
+    }
```

Comment:
Is `onMasterOnly` needed, if this is the default?   In what case would a user need to use `onMasterOnly`?